src/API.jl

```diff
@@ -1469,11 +1481,13 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
                         end
                         loaded && (n_loaded += 1)
                     catch err
-                        close(iob)
+                        # close pipes to end the std output monitor
+                        close(stderr_pipe)
+                        close(stdout_pipe)
```

Comment:
```suggestion                        
```
This introduces a data-race into your design, and suggests that the flow of events needs to be better managed so that they aren't happening from conflicting different places.